code/modules/integrated_electronics/subtypes/lists.dm

```diff
@@ -335,18 +363,66 @@
 		if(list_index > input_list.len)
 			set_pin_data(IC_OUTPUT, i, null)
 		else
-			set_pin_data(IC_OUTPUT, i, input_list[list_index])
+			var/data = input_list[list_index]
+			set_pin_data(IC_OUTPUT, i, data)
+			set_pin_data(IC_OUTPUT, i + number_of_pins, input_list[data])
 
 	push_data()
 	activate_pin(2)
 
 /obj/item/integrated_circuit/lists/deconstructor/small
 	name = "list deconstructor"
-	desc = "This circuit will write first four entries of input list, starting with index, into the output values."
+	desc = "This circuit will write first four entries of input list, starting with index, into the output values. Supports associative lists."
 	icon_state = "deconstr"
 	number_of_pins = 4
 
 /obj/item/integrated_circuit/lists/deconstructor/medium
 	name = "medium list deconstructor"
-	desc = "This circuit will write first eight entries of input list, starting with index, into the output values."
+	desc = "This circuit will write first eight entries of input list, starting with index, into the output values. Supports associative lists."
 	number_of_pins = 8
+
+/obj/item/integrated_circuit/lists/json_encode
+	name = "JSON encoder"
+	desc = "This circuit encodes a list into JSON format. Supports associative lists."
+	inputs = list("list" = IC_PINTYPE_LIST)
+	outputs = list("json" = IC_PINTYPE_STRING)
+	spawn_flags = IC_SPAWN_DEFAULT | IC_SPAWN_RESEARCH
+
+/obj/item/integrated_circuit/lists/json_encode/do_work()
+	pull_data()
+	var/list/L = get_pin_data(IC_INPUT, 1)
+	set_pin_data(IC_OUTPUT, 1, json_encode(L))
```

Comment:
json_encode can't deal with weakrefs.possibly needs special encoding.

Reply:
added a desc note to deal with it, if they do it anyways they'll just get a string instead of an object and their circuit doesn't work oh well so sad